### PR TITLE
Remove usage of legacy Optional syntax

### PR DESCRIPTION
**Pull request by sjakobi**
…and remove some related cruft.

### CODE REVIEW

**Comment by sjakobi** [comment]
I was wondering about the relationship between `dhall-json/examples/travis.dhall` and `dhall-json/examples/travis.yaml`. Are they supposed to represent the same config?

I'd add a  test then to ensure their equivalence.

File: dhall-json/src/Dhall/JSON.hs [hunk 1]
```diff
@@ -54,15 +54,15 @@
 
     Dhall @List@s translate to JSON lists:
 
-> $ dhall-to-json <<< '[1, 2, 3] : List Integer'
+> $ dhall-to-json <<< '[+1, +2, +3] : List Integer'
```
Comment (Gabriel439) [inline dhall-json/src/Dhall/JSON.hs]:
```suggestion
> $ dhall-to-json <<< '[1, 2, 3] : List Natural'
```

File: dhall-json/src/Dhall/JSON.hs [hunk 2]
```diff
@@ -54,15 +54,15 @@
...
 > [1,2,3]
 
     Dhall @Optional@ values translate to @null@ if absent and the unwrapped
     value otherwise:
 
-> $ dhall-to-json <<< '[] : Optional Integer'
+> $ dhall-to-json <<< 'None Integer'
```
Comment (Gabriel439) [inline dhall-json/src/Dhall/JSON.hs]:
```suggestion
> $ dhall-to-json <<< 'None Natural'
```

File: dhall-json/src/Dhall/JSON.hs [hunk 3]
```diff
@@ -54,15 +54,15 @@
...
 > null
-> $ dhall-to-json <<< '[1] : Optional Integer'
+> $ dhall-to-json <<< 'Some +1'
```
Comment (Gabriel439) [inline dhall-json/src/Dhall/JSON.hs]:
```suggestion
> $ dhall-to-json <<< 'Some 1'
```

File: dhall-nix/src/Dhall/Nix.hs [hunk 4]
```diff
@@ -45,10 +45,10 @@
     `Optional` values translate to @null@ if missing or the unwrapped value if
     present:
 
-> $ dhall-to-nix <<< "[] : Optional Integer"
+> $ dhall-to-nix <<< "None Integer"
```
Comment (Gabriel439) [inline dhall-nix/src/Dhall/Nix.hs]:
```suggestion
> $ dhall-to-nix <<< "None Natural"
```

File: dhall-nix/src/Dhall/Nix.hs [hunk 5]
```diff
@@ -45,10 +45,10 @@
...
 > null
 
-> $ dhall-to-nix <<< "[1] : Optional Integer"
+> $ dhall-to-nix <<< "Some +1"
```
Comment (Gabriel439) [inline dhall-nix/src/Dhall/Nix.hs]:
```suggestion
> $ dhall-to-nix <<< "Some 1"
```

**Comment by DediGithubJoint** [comment]
☺

**Pull request merged**